### PR TITLE
Handle Gumbel ppf endpoint domains

### PR DESCRIPTION
**Pull request by KirtiRamchandani**
## Summary

Handle closed probability-domain endpoints for `jax.scipy.stats.gumbel_l.ppf` and `gumbel_r.ppf`, and return `nan` for invalid scales.

Addresses the Gumbel `ppf` portion mentioned in #38318.

## Root cause

Both implementations used a strict `0 < p < 1` mask before returning the inverse-CDF formula. That made `p == 0` and `p == 1` return `nan`, while SciPy returns the support endpoints. The same mask also did not enforce the positive-scale domain for `ppf`, unlike the neighboring Gumbel functions.

## Solution

- Accept `0 <= p <= 1` when `scale > 0`
- Return `loc + scale * -inf` for `p == 0`
- Return `loc + scale * inf` for `p == 1`
- Preserve `nan` for out-of-domain probabilities and invalid scales

## Tests

- `python -m pytest tests/scipy_stats_test.py -k "testGumbelPpfDomain or testGumbelPpfInvalidScale" -q`
- `python -m pytest tests/scipy_stats_test.py -k "testGumbelRPpf or testGumbelLPpf" -q`
- `python -m pytest tests/scipy_stats_test.py -k Gumbel -q`
- `python -m ruff check jax/_src/scipy/stats/gumbel_l.py jax/_src/scipy/stats/gumbel_r.py tests/scipy_stats_test.py`
- `git diff --check`

### CODE REVIEW

**Comment by jakevdp** [comment]
This falls under #38651; please note that it may take a while for this to receive attention from the team.
